src/main/java/homework_5/power_of_number/PowerOfNumber.java

```diff
@@ -0,0 +1,54 @@
+package homework_5.power_of_number;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.math.BigInteger;
+
+public class PowerOfNumber {
```

Comment:
Not a common implementation of recursion, it can be confusing. Well, works anyway